### ISSUE TITLE
10.2.0 hydrate strange behavior using JSDOM

### ISSUE THREAD

**Issue by jchip** [preactjs/preact#2237]
Hi, I have the following component:

```js
/** @jsx h */
import { h } from "preact"
export default props => <p>hello {props.foo}</p>
```

babel would transpile that to

```js
props => h("p", null, "hello ", props.foo)
```

I have some unit test using JSDOM to run with node.js, like this:

```js
const { h, hydrate } = require("preact");
const { JSDOM } = require("jsdom");

const dom = new JSDOM(`<div id="test"><p>hello bar</p></div>`);
global.window = dom.window;
global.document = dom.window.document;
const element = dom.window.document.getElementById("test");

// component as transpiled by babel
const Component = props => h("p", null, `hello `, props.foo);

hydrate(h(Component, { foo: "bar" }), element);
console.log(element.innerHTML);
```

- With preact 10.2.0, the output from above is `<p>hello barbar</p>`, which is unexpected.

- With preact 10.1.1, the output is `<p>hello bar</p>`

note that if I changed the component to:

```js
const Component = props => h("p", null, `hello ${props.foo}`);
```

then 10.2.0 behaves the same as 10.1.1.

**Comment by JoviDeCroock** [comment]
Hey :wave:,

Thanks for the accurate reproduction, it enabled me to quickly put up a PR with a fix!

**Comment by marvinhagemeister** [comment]
@jchip Just published Preact `10.2.1` which includes the fix for this issue :+1:
